crates/harness/src/monolith/behavior.rs

```diff
@@ -0,0 +1,172 @@
+use super::*;
+
+/// Defines how an emulated monolith should behave in response to messages from the balancer.
+pub trait Behavior: Send {
+    /// Processes the given message from the balancer, and returns messages to send back to the balancer.
+    fn on_msg(&self, msg: &MsgB2M, state: &mut MonolithState) -> Vec<MsgM2B>;
+}
+
+macro_rules! impl_behavior_tuple {
+    ($($ty:ident),*) => {
```

Comment:
$($ty:ident) ?????????

Reply:
This is macro syntax. `$ty:ident` means that the token it's looking for when parsing should be an identifier, and the `,*` makes it so that you can specify multiple separated by commas.